utils/model.py

```diff
@@ -0,0 +1,70 @@
+import torch
+import torch.nn as nn
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+class BaseModel(nn.Module, ABC):
+    @abstractmethod
+    def forward(self, x):
+        pass
+
+    @abstractmethod
+    def fit(self, train_loader, val_loader, num_epochs, learning_rate, device):
```

Comment:
We may make both `fit` and `predict` methods more flexible, which means they take no arguments.

```py
def fit(self):
    torch.save("i_know_this_path_because_i_have_set_it.pt")

def predict(self):
    weights = torch.load("i_know_this_path_because_i_have_set_it.pt")
```